pages/uploader.tsx

```diff
@@ -1,150 +1,268 @@
+        name: acceptedFiles[0].name.replace(/\.[^/.]+$/, ''),
+        errors: [],
+      };
+      setFile(mappedAccepted);
+      //   },
+      //   false
+      // );
+      // const mappedAccepted2 = acceptedFiles.map((file) => ({
+      //   file,
+      //   errors: [],
+      // }));
+      // setFile((curr) => [...curr, mappedAccepted, ...rejectedFiles]);
+    },
+    []
+  );
+
+  const { getRootProps, getInputProps } = useDropzone({
+    onDrop,
+    multiple: false,
+    accept: ['audio/*'],
+  });
+
+  function uploadFile(file: File) {
+    const sermonRef = ref(storage, `sermons/${file.name}`);
+    const uploadTask = uploadBytesResumable(sermonRef, file);
+    const sermonData = {
+      title: title,
+      subtitle: subtitle,
+      date: date,
+      description: description,
+      speaker: speaker,
+      scripture: scripture,
+      topic: topic.split(' '),
+      ref: sermonRef.fullPath,
+    };
+    uploadTask.on(
+      'state_changed',
+      (snapshot) => {
+        const progress =
+          (snapshot.bytesTransferred / snapshot.totalBytes) * 100;
+        setUploadProgress(`${Math.round(progress)}%`);
+        switch (snapshot.state) {
+          case 'paused':
+            break;
+          case 'running':
+            break;
+        }
+      },
+      (error) => {
+        setUploadProgress(`Error: ${error}`);
+      },
+      () => {
+        // With autogenerated id
+        addDoc(collection(db, 'sermons'), sermonData);
+
+        // with custom id so it doesnt duplicate upload
+        // setDoc(doc(db, 'sermons', title), sermonData);
+
+        setUploadProgress('Uploaded!');
+      }
+    );
+
+    // console.log('Upload File', file);
+    // const url = '...';
+    // return await new Promise((res, rej) => {
+    //   const xhr = new XMLHttpRequest();
+    //   xhr.open('POST', url);
+    //   xhr.onload = () => {
+    //     res('00')
+    //   }
+    //   xhr.onerror = (evt) => rej(evt);
+    //   xhr.upload.onprogress = (event) => {
+    //     if (event.lengthComputable) {
+    //       const percentage = (event.loaded / event.total) * 100;
+    //       onProgress(Math.round(percentage))
+    //     }
+    //   };
+    //   const formData = new FormData();
+    //   formData.append('file', file);
+    //   formData.append('key', key)
+    //   xhr.send(formData)
+    // });
+  }
+
+  return (
+    <div className={styles.container}>
+      <Navbar />
+      <h1>Uploader</h1>
+      <form>
+        <label>
+          Name:
+          <input
+            type="text"
+            value={title}
+            onChange={(e) => setTitle(e.target.value)}
+          />
+        </label>
+        <label>
+          Subtitle:
+          <input
+            type="text"
+            value={subtitle}
+            onChange={(e) => setSubtitle(e.target.value)}
+          />
+        </label>
+        <label>
+          Date:
+          <input
+            type="text"
```

Comment:
```suggestion
            type="date"
```

Reply:
On-top of the ui, the date is being saved as a string in firestore, we should save it as a timestamp